src/parse/matches/arg_matches.rs

```diff
@@ -331,17 +331,21 @@ impl ArgMatches {
     pub fn value_of_t<R>(&self, name: &str) -> Result<R, Error>
     where
         R: FromStr,
-        <R as FromStr>::Err: Display,
+        <R as FromStr>::Err: Display + Debug,
     {
         if let Some(v) = self.value_of(name) {
-            v.parse::<R>().map_err(|e| {
-                Error::value_validation_auto(&format!(
+            let parsed = v.parse::<R>();
+
+            if let Err(e) = parsed {
+                Err(Error::value_validation_auto(&format!(
                     "The argument '{}' isn't a valid value: {}",
                     v, e
-                ))
-            })
+                ))?)
+            } else {
+                Ok(parsed.expect(INTERNAL_ERROR_MSG))
+            }
```

Comment:
You can use `match` instead of `if let Err {} else { unwrap }`. 

Btw, you can just `.map_err` here.

Reply:
`map_err` doesn't work because `Error::value_validation_auto` returns a `Result` and using the `?` gives:

```
the `?` operator can only be used in a closure that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
```

Reply:
Than use [`Result::or_else`](https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else) which is made exactly for this situation.

Reply:
I kind of think this situation when *creating* an error - not *displaying*! - can return `io::Error` is rather unfortunate. I''' be seeing into this further, but not blocker for this PR.